tspec/base/posix/posix/errno.h.ts

```diff
@@ -3,8 +3,7 @@
 #
 # See doc/copyright/ for the full copyright terms.
 
-
-+EXP (extern) int errno ;
++IMPLEMENT "c/c89", "errno.h.ts" ;
```

Comment:
This is really interesting. If there's a difference from C89 here, it is likely to be because this version of POSIX did actually specify errno that way. I'll try to check this. Thank you!

Reply:
C89 §4.1.3 p2:
> and `errno` which expands to a modifiable lvalue/83/ that has type `int`
so our C89 definition is correct (of course)

I can't find a copy of the 1988 posix text, IEEE Std 1003.1-1988. But I can find 1990, which does indeed say:
```
extern int errno;
```
with no possibility for a macro.

So we cannot `+IMPLEMENT` c89's definition here, unfortunately.

Instead I think the proper thing to do, is to mark `errno` as __WRONG in the osdep stuff for Linux systems where this is a macro (i.e. all of them).

I also see that the tspec definitions for posix1 `+IMPLEMENT` c89, which I now believe to be incorrect.

Perhaps we have the option to provide some compatibility glue in the osdep/ hacked includes, such that our `posix` and `posix1` TDF tokens are somehow mapped to the underlying libc macro. I'm not sure exactly how to do that, but maybe with PL_TDF.

If we can't do that, this is a great shame. In practice that means we'll need to depend on whatever version of `posix` for rebuild did introduce the macro form, not on `posix`. Or we find a way to drop the dependency there.

Reply:
cc #11 and #7

Reply:
From what I understood from the C definition is that up until C11, it was allowed to be either an assignable thing or a macro.  (http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1338.htm) While the posix implementations mostly had macros for a very long time.

However, the problem not so much lies in the definition, but the result of the actual code.

Because `+EXP (extern) int errno` causes an actual implementation in the header file that is output, as opposed to the `+EXP lvalue int errno;` in c89 that only causes a #pragma.
So in the case of using the c89 header, it will allow the "actual" errno.h implementation of the system have ownership of the symbols.
While in the original posix spec, the errno symbol is reimplemented as `extern int errno ;` in the resulting `errno.h`.
It would be fine if the spec would just output an appropriate pragma.

Reply:
From irc:
> errno is an lvalue equivalent to extern int, issue 5 made changes to allow for a thread local  errno more of which was done in issue 6 with issue 7 completing the task and making it no longer potentially different from c90.